kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -207,12 +210,11 @@ private static GenericKubernetesResource filterResourceFromRef(Clusters cluster,
     }
 
     private static VirtualCluster getVirtualCluster(KafkaProxy primary,
-                                                    Clusters cluster) {
-        var o = ResourcesUtil.distinctClusters(primary);
-        var clusterNum = o.indexOf(cluster);
-
+                                                    VirtualKafkaCluster cluster,
+                                                    int clusterNum) {
+        String bootstrap = cluster.getSpec().getTargetCluster().getBootstrapping().getBootstrapAddress();
         return new VirtualCluster(
-                new TargetCluster(cluster.getUpstream().getBootstrapServers(), Optional.empty()),
+                new TargetCluster(bootstrap, Optional.empty()),
```

Comment:
It would be really nice if we could just pass the `bootstraping` node in here rather than having to unpack it a single address.

Reply:
`TargetCluster` here is `io.kroxylicious.proxy.config`, out of the operators ken